pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -111,17 +130,97 @@ func (r *Reconciler) ValidateClusterSpec(ctx context.Context, log logr.Logger, t
 	return controller.Result{}, nil
 }
 
-// ReconcileControlPlane applies the control plane CAPI objects to the cluster.
-func (r *Reconciler) ReconcileControlPlane(ctx context.Context, log logr.Logger, tinkerbellScope *Scope) (controller.Result, error) {
+// GenerateSpec generates Tinkerbell control plane and workers spec.
+func (r *Reconciler) GenerateSpec(ctx context.Context, log logr.Logger, tinkerbellScope *Scope) (controller.Result, error) {
 	spec := tinkerbellScope.ClusterSpec
-	log = log.WithValues("phase", "reconcileControlPlane")
-	log.Info("Applying control plane CAPI objects")
+	log = log.WithValues("phase", "GenerateSpec")
+	log.Info("Generating tinkerbell control plane and workers spec")
+
 	cp, err := tinkerbell.ControlPlaneSpec(ctx, log, clientutil.NewKubeClient(r.client), spec)
 	if err != nil {
 		return controller.Result{}, err
 	}
+	tinkerbellScope.ControlPlane = cp
+
+	w, err := tinkerbell.WorkersSpec(ctx, log, clientutil.NewKubeClient(r.client), spec)
+	if err != nil {
+		return controller.Result{}, errors.Wrap(err, "generating workers spec")
+	}
+	tinkerbellScope.Workers = w
 
-	return clusters.ReconcileControlPlane(ctx, r.client, toClientControlPlane(cp))
+	return controller.Result{}, nil
+}
+
+// DetectOperation detects change type.
+func (r *Reconciler) DetectOperation(ctx context.Context, log logr.Logger, tinkerbellScope *Scope) (controller.Result, error) {
+	log = log.WithValues("phase", "DetectOperation")
+	log.Info("Detecting operation type")
```

Comment:
Small suggestion that minimizes regurgitation in the log event. This can be applied to all the phase begining logs we have.

```suggestion
	log.Info("Begin phase")
```

Reply:
IMO I think we dont need this log here? We dont usually add any begin and end logs in other phases.